src/test/scala/uk/gov/nationalarchives/tdr/api/db/repository/FFIDMetadataRepositorySpec.scala

```diff
@@ -83,13 +83,93 @@ class FFIDMetadataRepositorySpec extends AnyFlatSpec with TestDatabase with Scal
     TestUtils.createFile(UUID.fromString(fileThreeId), consignmentId)
     TestUtils.createFile(UUID.fromString(fileFourId), consignmentId)
 
-    (1 to 7).foreach { _ => TestUtils.addFFIDMetadata(fileOneId)}
+    TestUtils.addFFIDMetadata(fileOneId)
+    TestUtils.addFFIDMetadata(fileTwoId)
+    TestUtils.addFFIDMetadata(fileThreeId)
```

Comment:
Ah, yes, because I noticed the same test in the FileMetadata Repository and thought that it was a just copied over by mistake, because I was wondering why you would you call addFFIDMetadata multiple times on a file, but I guess it could happen and probably might since the fileId doesn't have to be unique which also explains why we have a `groupBy` in the "count" slick query. 

1. Are we allowing multiple FFID metadata for versioning purposes (i.e. "this file was updated with a new FFID on this date, originally it was X")? 
      i. Could this same logic apply to the File metadata table? Could you/would you want to have multiple client-side checksum rows? Maybe if the process of updating the database fails and you need to start again? 
2. Or can a file have multiple file format identifications for a single file?

Reply:
It's a good question, and one we don't really have an answer to yet.

I can only think of two ways that you might end up with multiple FFID rows for the same file:

1. There's some network glitch when the api-update lambda saves the metadata to the consignment API which leads to the data being saved but the api-update lambda not receiving the successful response. The api-update lambda tries up to three times to save the data, so it will assume the first one has failed and try again. In this case, the data in each row will be identical.
2. We re-trigger the file check messages for a file, even though the initial file checks succeeded. We don't have a mechanism for doing this at the moment, but I suspect we'll need to build one at some point. In this case, the data could potentially be different if, for example, we upgraded DROID in between running the first checks and the second.

So there is a bit of ambiguity here which we should definitely sort out at some point, but I don't think we have enough information to know what we _should_ do here. It will depend on what information we want to keep for audit, and on what we think we'll do when we let users replace specific files (e.g. letting them overwrite a password-protected file with the unlocked file). So I'd rather tackle this if we end up in situation 2 a lot, or when we start tackling file replacement.

In the meantime, I think it's worth keeping the old test logic which saves some duplicate FFID so that we understand what happens when there's duplicate data, even though we don't think it will happen very often.